presto-hive/src/main/java/io/prestosql/plugin/hive/S3SelectPushdown.java

```diff
@@ -102,7 +100,7 @@ public static boolean isCompressionCodecSupported(InputFormat<?, ?> inputFormat,
 
     private static boolean areColumnTypesSupported(List<Column> columns)
     {
-        if (columns == null || columns.isEmpty()) {
```

Comment:
Move this to a separate commit as it is the only change that isn't making utility classes final with a private constructor.